packages/blockchain/src/index.ts

```diff
@@ -939,30 +938,45 @@ export default class Blockchain implements BlockchainInterface {
         }
       }
 
-      // set total difficulty in the current context scope
-      if (this._headHeaderHash) {
-        currentTd.header = await this.getTotalDifficulty(this._headHeaderHash)
-      }
-      if (this._headBlockHash) {
-        currentTd.block = await this.getTotalDifficulty(this._headBlockHash)
+      if (block._common.consensusType() === ConsensusType.ProofOfStake) {
+        if (block._common.hardfork() !== 'merge') {
```

Comment:
As stated in the old PR, this check should not be there, see [comment](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1406/files#r689925951) for reference.

Reply:
Yep, I missed this when I cherrypicked the commit from the old PR.  Will be removed in my next commit.

Reply:
Yep, I missed this when I cherrypicked the commit from the old PR. Will be removed in my next commit.